crates/swift-bridge-build/src/package.rs

```diff
@@ -0,0 +1,287 @@
+	pub paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>,
+	/// The directory where the package will be saved
+	pub out_dir: &'a dyn AsRef<Path>,
+	/// The name for the Swift package
+	pub package_name: &'a str
+}
+
+impl<'a> GeneratePackageConfig<'a> {
+	/// Creates a new `GeneratePackageConfig` for generating Swift Packages from Rust code.
+	pub fn new(bridge_dir: &'a dyn AsRef<Path>, paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>, out_dir: &'a dyn AsRef<Path>, package_name: &'a str) -> Self {
+		Self {
+			bridge_dir,
+			paths,
+			out_dir,
+			package_name
+		}
+	}
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Eq, PartialEq, Hash)]
+/// Currently supported platforms for genereting Swift Packages.
+pub enum ApplePlatform {
+	/// `aarch64-apple-ios`
+	iOS,
+	/// `x86_64-apple-ios`
+	/// `aarch64-apple-ios-sim`
+	///
+	/// iOS simulator for debugging in XCode's simulator.
+	Simulator,
+	/// `x86_64-apple-darwin`
+	macOS,
+	/// no official Rust target for this platform
+	MacCatalyst,
+	/// `aarch64-apple-tvos`
+	/// `x86_64-apple-tvos`
+	tvOS,
+	/// no official Rust target for this platform
+	watchOS,
+	/// no official Rust target for this platform
+	watchOSSimulator,
+	/// no official Rust target for this platform
+	carPlayOS,
+	/// no official Rust target for this platform
+	carPlayOSSimulator,
+}
+
+impl ApplePlatform {
+	/// The directory name inside of the xcframework for the specified platform.
+	fn dir_name(&self) -> &str {
+		match self {
+			ApplePlatform::iOS => "ios",
+			ApplePlatform::Simulator => "simulator",
+			ApplePlatform::macOS => "macos",
+			ApplePlatform::MacCatalyst => "macos-mac-catalyst",
+			ApplePlatform::tvOS => "tvos",
+			ApplePlatform::watchOS => "watchos",
+			ApplePlatform::watchOSSimulator => "watchos-simulator",
+			ApplePlatform::carPlayOS => "carplay",
+			ApplePlatform::carPlayOSSimulator => "carplay-simulator",
+		}
+	}
+	
+}
+
+/// Generates an xcframework embedded in a Swift Package from the Rust project.
+///
+/// - Also see the [relevant book chapter](https://chinedufn.github.io/swift-bridge/building/swift-packages/index.html)
+/// 
+/// # Parameters
+/// - `config`: The config for generating the swift package, contains the directory
+///    containing te bridges, the paths to the libraries per platform and the output directory
+pub fn generate_package(config: GeneratePackageConfig) {
+	// Create output directory //
+	let output_dir: &Path = config.out_dir.as_ref();
+	if !&output_dir.exists() {
+		fs::create_dir_all(&output_dir).expect("Couldn't create output directory");
+	}
+	
+	// Generate XCFramework //
+	gen_xcframework(&output_dir, &config);
+	
+	// Generate Swift Package //
+	gen_package(&output_dir, &config);
+}
+
+/// Generates the XCFramework
+fn gen_xcframework(output_dir: &Path, config: &GeneratePackageConfig) {
+	// Create directories
+	let tmp_framework_path = std::env::temp_dir().join("swiftbridge._tmp_framework");
```

Comment:
Do we need to worry about the fact that `std::env::temp_dir` isn't secure?

I don't have any thoughts off the top of my head, just checking.